zerver/tests/test_messages.py

```diff
@@ -5214,3 +5215,22 @@ def test_bulk_get_huddle_user_ids(self) -> None:
 
     def test_bulk_get_huddle_user_ids_empty_list(self) -> None:
         self.assertEqual(bulk_get_huddle_user_ids([]), {})
+
+class NoRecipientIDsTest(ZulipTestCase):
+    def test_no_recipient_ids(self) -> None:
+        # Takes code coverage for stream_recipients.py to 100%
+        # and checks that having no recipient ids
+        # returns nothing.
+
+        user_profile = self.example_user('cordelia')
+        # self.login_user(user_profile)
+
+        Subscription.objects.all().delete()
```

Comment:
Let's just delete the subscriptions to streams from the user; it's cheaper and says what we mean.

And can you verify that if we remove that conditional in `stream_recipient.py`, this test fails?

Reply:
I tried deleting the subscriptions to streams but I got an "Manager does not have attribute object" error. I don't see Subscription.objects.delete() used anywhere in Zulip.

Reply:
I did indeed get an error in process_query after removing the if statement. This confirms the functionality of the test.
![image](https://user-images.githubusercontent.com/35404953/80045090-4fc0af00-84bb-11ea-8655-7a8ee78f525a.png)

Reply:
Ahh, I meant `.filter(user_profile=user_profile, recipient__type=Recipient.STREAM).delete()`; I'd forgotten `.delete()` doesn't accept filtering kwargs.